### DISCUSSION TITLE
Docker service error - l4t-32.5.0 with Mender

### DISCUSSION THREAD

**Discussion by tirankas** [OE4T/tegra-demo-distro#75] [General]
Hello,
I have made a demo full image (docker, nvidia runtime, opencv etc) with Mender support enabled for Jetson NX Xavier.
The image seems to powerup fine but I have a problem with the docker support.

# systemctl status docker
* docker.service - Docker Application Container Engine
     Loaded: loaded (/lib/systemd/system/docker.service; enabled; vendor preset>
     Active: inactive (dead)
TriggeredBy: * docker.socket
       Docs: https://docs.docker.com

# docker ps
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?

Anyone familiar with this error?

**Comment by madisongh** [comment]
Can you include the output from `journalctl -b -u docker.service`? The log should tell you why it's failing to start.

**Reply by tirankas** [reply]
Nothing there:
journalctl -b -u docker.service
-- Logs begin at Sun 2020-09-20 10:43:59 UTC, end at Sun 2020-09-20 10:44:28 UT>
-- No entries --

**Reply by madisongh** [reply]
OK, so it hasn't even started yet.  Is there any status for the socket unit --   `systemctl status docker.socket`  ?

**Reply by tirankas** [reply]
systemctl start docker.socket
[   98.030591] overlayfs: failed to resolve '/data/overlays/home-upper': -2
[   98.036300] overlayfs: failed to resolve '/data/overlays/var-lib-upper': -2
Job for docker.socket canceled.

**Reply by ichergui** [reply]
Hmm, please check the systemd service `data-overlay-setup.service` and `fstab`  ?

**Reply by tirankas** [reply]
Thanks, I found the problem, am building with the fix.
The problem was the data overlay setup recipe was linked with a wrong distro (I renamed my distro and forgot to rename the distro dependencies.).

Thanks again, will post back after testing.

**Reply by madisongh** [reply]
If `/var/lib` and `/home` aren't getting mounted properly, you should be seeing more service failures than just docker. Did you make any changes to the distro setup?  I just booted an image on my Xavier NX devkit built off `dunfell-l4t-r32.5.0`, and everything came up as expected.

The `/data` partition should be mounted, read-write, on `/dev/mmcblk0p12`. Overlay mounts should also be there for `/etc/docker`, `/var/lib`, and `/home`.  If they aren't present, as @ichergui mentions, that should be investigated, since several services need to have writable storage, while with Mender we have a read-only rootfs.

**Reply by tirankas** [reply]
Hi Matt,
Yes, this is my own distro (it's name is not tegrademo-mender) but forgot to rename places that depends on the tegrademo-mender distro.

Thanks

**Reply by tirankas** [reply]
Issue solved, thanks for the help

**Reply by dwalkes** [reply]
Thanks @tirankas 

>  but forgot to rename places that depends on the tegrademo-mender distro.

Is this some change we should make to documentation or https://github.com/OE4T/tegra-demo-distro/blob/master/scripts-setup/create-distro-layer ?

**Reply by tirankas** [reply]
Hi Dan,
No, this is not really related.
It totally on my side.
I have renamed the distro name manually but didn't modify the append_distro macros.
Since I am now done with all my changes and everything work fine including capturing & analytics, I will get to the point of proposing changes to the manual, making sure everything is crystal clear...
